src/app/components/core/system-alert/system-alert.component.ts

```diff
@@ -58,6 +72,49 @@ export class SystemAlertComponent {
     );
   }
 
+  sortByTimestamp() {
+    // toggle the next possible button text
+    this.sortByKey = 'urgency';
+
+    return this.systemAlerts.sort((a, b) => new Date(b.timestamp).getTime() - new Date(a.timestamp).getTime());
+  }
+
+  groupByUrgencySortedByTime(alerts = []) {
+    // don't need to pass alerts array as param when clicking on 'Sort by Urgency' link, so assign "this.systemAlerts" to alerts variable in that case
+    if (alerts.length === 0) {
+      alerts = this.systemAlerts;
+    }
+    const urgencyOrder = { "Critical": 0, "High": 1, "Normal": 2, "Low": 3 };
+    
+    // Group by urgency
+    const groupedByUrgencyObj = alerts.reduce((alert, item) => {
+      if (!alert[item.urgency]) {
+        alert[item.urgency] = [];
+      }
+      alert[item.urgency].push(item);
+      return alert;
+    }, {});
+
+    // array of alerts Group by sorted urgency
+    let groupedByUrgency = Object.keys(groupedByUrgencyObj)
+    .sort((a, b) => urgencyOrder[a] - urgencyOrder[b])
+    .map((key) => 
+    [...groupedByUrgencyObj[key]]
+    );
+
+    // Sort each group by timestamp
+    groupedByUrgency.forEach(alerts => {
+      alerts.sort((a, b) => new Date(b.timestamp).getTime() - new Date(a.timestamp).getTime());
+    });
+    
+    const systemAlerts = groupedByUrgency.reduce((a, value) => a.concat(value), []);
```

Comment:
What is the use of this? and can be directly assigned to  this.systemAlerts = ... if required? Add an inline comment